src/main/webapp/table.js

```diff
@@ -0,0 +1,149 @@
+/* Class representing Table, used for sorting and rendering.*/
+//TODO: Check which additional functions need to be bound in constructor
+class Table {
+  constructor(dataTable, name, colSchemas, id) {
+    this.name = name;
+    this.colSchemas = colSchemas;
+    this.id = id;
+    this.dataTable = new Array(dataTable.length); // Number rows
+    this.sortDirections = new Array(colSchemas.length); // Number cols
+    this.makeTableWithTypes(dataTable);
+
+    this.setTable = this.setTable.bind(this);
+  }
+
+  getName() {
+    return this.name;
+  }
+
+  getDataTable() {
+    return this.dataTable;
+  }
+
+  getDataType(colIndex) {
+    const colSchema = this.colSchemas[colIndex];
+    return colSchema.schemaType;
+  }
+
+  // Initialize table with actual column types
+  makeTableWithTypes(dataTable) {
+    let rowIndex = 0;
+    for (rowIndex in dataTable) {
+      const row = dataTable[rowIndex];
+      this.dataTable[rowIndex] = new Array(row.length);
+  
+      let col;
+      for (col in row) {
+        const cell = row[col];
+        const type = this.getDataType(col);
+        if (cell == "NULL") {
+          this.dataTable[rowIndex][col] = "";
+        } else {
+          if (type == "INT64") {
+            const cellToInt = parseInt(cell);
+            this.dataTable[rowIndex][col] = cellToInt;
+          } else {
+            this.dataTable[rowIndex][col] = cell;
+          }
+        }
+      }
+    }
+  }
+
+  fetchTable() {
+    setTimeout(function() {
+      this.renderTable();
+    }.bind(this), 0);
+  }
+
+  // 0 is ascending, 1 is descending -- start everything ascending
+  initSortDirectionsArray() {
+    let index = 0;
+    for (index in this.colSchemas.length) {
+      this.sortDirections[index] = 0;
+    }
+  }
+
+  flipSortDirection(index) {
+    let currentDirection = this.sortDirections[index];
+    if (currentDirection == 0) {
+      this.sortDirections[index] = 1;
+    } else {
+      this.sortDirections[index] = 0;
+    }
+  }
+
+  getSortDirection(index) {
+    return this.sortDirections[index];
+  }
+
+  sortRows() {
+    const newRows = this.rows.slice();
+    newRows.sort();
+    this.setRows(newRows);
+  }
+
+  setTable(dataTable) {
+    this.dataTable = dataTable;
+  }
+
+  remove() {
+    document.getElementById("table_" + this.name).remove();
+    document.getElementById("header_" + this.name).remove();
+  }
+
+  renderTable() {
```

Comment:
This function will be very important for our pagination and search bar functionality, so we should try to preserve this as much as possible

Reply:
yep